server/src/board/tile/rail/decouplerrailtile.cpp

```diff
@@ -67,6 +67,11 @@ DecouplerRailTile::DecouplerRailTile(World& world, std::string_view _id)
 
   Attributes::addObjectEditor(deactivate, false);
   m_interfaceItems.add(deactivate);
+
+  outputMap->onOutputStateMatchFound.connect([this](DecouplerState value)
+    {
+      setState(value, true);
```

Comment:
You can just do `state.setValueInternal(value);` here, because it is an internal update.

Reply:
Yes but if in the future other operations are added inside `setState()` it might be worth calling them too.
`setValueInternal()` is more of an implementation design of decoupler in my opinion.

Reply:
ok, that could be possible indeed, then add a private/protected `setStateInternal()` that has the bypass option. The public one shouldn't have a bypass option.

Reply:
`DecouplerRailTile::setState()` is already private actually. Should I rename it?

Reply:
No, my bad, I confused it with `setPosition` of the Turnouts, that one is public because of the scripting engine.
Just leav it like this (for now at least ;))